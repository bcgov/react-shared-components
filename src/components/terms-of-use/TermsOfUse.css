.scroll-box {
  overflow: auto;
  width: 100%;
  height: 500px;
  border: 1px solid;
  padding: 15px;
  margin-bottom: 30px;
}

.full-width {
  width: 100%;
}

.print-page {
  cursor: pointer;
  text-decoration: underline;
  color: #1a5a96;
  float: right;
  margin-top: 10px;
  margin-right: 15px;
}

@media print {
  .printable {
    overflow: visible;
    height: auto;
  }

  .non-printable {
    display: none;
  }
}
