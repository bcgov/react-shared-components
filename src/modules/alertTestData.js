import React from "react";
import { Alert } from "../components/alert/Alert";

export function generateAlert(icon, type, styling, element) {
  return <Alert icon={icon} type={type} styling={styling} element={element} />;
}
