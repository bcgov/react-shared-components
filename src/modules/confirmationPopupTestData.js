/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getConfirmationPopupTestData() {
  return { body };
}
