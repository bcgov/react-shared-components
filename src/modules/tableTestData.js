const elements = [
  { name: "BCeID", value: "bobross42" },
  {
    name: "Last Name",
    value: "Ross",
  },
  { name: "First Name", value: "Bob" },
  { name: "Email Address", value: "bob.ross@example.com" },
];

export function getTableElementsTestData() {
  return elements;
}
