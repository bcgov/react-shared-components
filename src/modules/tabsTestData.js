import React from "react";

export const tabs = [
  {
    title: "Tab1",
    content: <p> Tab1 Content... </p>,
  },
  {
    title: "Tab2",
    content: <p> Tab2 Content... </p>,
  },
  {
    title: "Tab3",
    content: <p> Tab3 Content... </p>,
    isDisabled: true,
  },
];
