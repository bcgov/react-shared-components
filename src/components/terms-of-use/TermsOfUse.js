import React from "react";
import PropTypes from "prop-types";
import { FaPrint } from "react-icons/fa";

import "./TermsOfUse.css";

export const TermsOfUse = ({ acceptTerms, content, heading, confirmText }) => {
  return (
    <div>
      <div className="non-printable full-width">
        <span
          role="button"
          className="print-page print"
          tabIndex={0}
          onClick={() => window.print()}
          onKeyDown={() => window.print()}
        >
          Print
          <FaPrint style={{ marginLeft: "10px" }} />
        </span>

        <h3>{heading}</h3>
      </div>

      <section className="scroll-box printable">{content}</section>

      <section className="non-printable pt-2">
        <label htmlFor="acceptTerms">
          <input id="acceptTerms" type="checkbox" onClick={acceptTerms} />
          &nbsp;
          <b>{confirmText}</b>
          <span id="asterisk" className="mandatory">
            *
          </span>
        </label>
      </section>
    </div>
  );
};

TermsOfUse.propTypes = {
  acceptTerms: PropTypes.func.isRequired,
  content: PropTypes.element.isRequired,
  heading: PropTypes.string.isRequired,
  confirmText: PropTypes.string.isRequired,
};
