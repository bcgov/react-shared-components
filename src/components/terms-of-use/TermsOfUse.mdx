import { Meta, Story } from "@storybook/addon-docs";
import { TermsOfUse } from "./TermsOfUse";

<Meta title="TermsOfUse" component={TermsOfUse} />

# Terms Of Use

The Terms of Use, or sometimes known as the Electronic Service Agreement, are used when you
need to present terms/conditions to the user which they must be aware of and consent to
prior to proceeding with their process in your application.

## API

| Prop        | Type     | Required | Description                                               | Default |
| ----------- | -------- | -------- | --------------------------------------------------------- | ------- |
| acceptTerms | function | true     | describes what happens when user accepts the terms of use |         |
| content     | element  | true     | jsx representing the content of the terms of use          |         |
| heading     | string   | true     | value of the heading for the terms of use                 |         |
| confirmText | string   | true     | text for the accept terms section                         |         |

<br />

## Basic Terms of Use

The typical terms of use, with the ability to print the terms.

<Story name="Terms of Use" id="terms-of-use--default"></Story>

```bash
const heading = "Terms of Use";

const confirmText = "I accept these terms and conditions";

const content = getTestData();

<TermsOfUse
  acceptTerms={action("accept terms of use")}
  content={content}
  heading={heading}
  confirmText={confirmText}
/>
```
