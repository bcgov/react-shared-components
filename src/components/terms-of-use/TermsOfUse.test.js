import React from "react";
import { render, fireEvent, getByText, waitFor } from "@testing-library/react";
import testBasicSnapshot from "../../TestHelper";
import { getTestData } from "../../modules/termsOfUseTestData";

import { TermsOfUse } from "./TermsOfUse";

describe("TermsOfUse Component", () => {
  window.print = jest.fn();
  const acceptTerms = jest.fn();

  const content = getTestData();

  const heading = "Terms of Use";

  const confirmText = "I accept these terms and conditions";

  test("Matches the snapshot", () => {
    const termsOfUse = (
      <TermsOfUse
        acceptTerms={acceptTerms}
        content={content}
        heading={heading}
        confirmText={confirmText}
      />
    );

    testBasicSnapshot(termsOfUse);
  });

  test("Print terms of use (click)", async () => {
    const { container } = render(
      <TermsOfUse
        acceptTerms={acceptTerms}
        content={content}
        heading={heading}
        confirmText={confirmText}
      />
    );

    fireEvent.click(getByText(container, "Print"));

    await waitFor(() => {
      expect(window.print).toHaveBeenCalled();
    });
  });

  test("Print terms of use (keydown)", async () => {
    const { container } = render(
      <TermsOfUse
        acceptTerms={acceptTerms}
        content={content}
        heading={heading}
        confirmText={confirmText}
      />
    );

    fireEvent.keyDown(getByText(container, "Print"));

    await waitFor(() => {
      expect(window.print).toHaveBeenCalled();
    });
  });
});
