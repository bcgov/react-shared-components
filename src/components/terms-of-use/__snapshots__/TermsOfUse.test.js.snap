// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsOfUse Component Matches the snapshot 1`] = `
<div>
  <div
    className="bcgov-non-printable bcgov-full-width"
  >
    <span
      className="bcgov-print-page print"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      Print
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
            "marginLeft": "10px",
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M448 192V77.25c0-8.49-3.37-16.62-9.37-22.63L393.37 9.37c-6-6-14.14-9.37-22.63-9.37H96C78.33 0 64 14.33 64 32v160c-35.35 0-64 28.65-64 64v112c0 8.84 7.16 16 16 16h48v96c0 17.67 14.33 32 32 32h320c17.67 0 32-14.33 32-32v-96h48c8.84 0 16-7.16 16-16V256c0-35.35-28.65-64-64-64zm-64 256H128v-96h256v96zm0-224H128V64h192v48c0 8.84 7.16 16 16 16h48v96zm48 72c-13.25 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
        />
      </svg>
    </span>
    <h3>
      Terms of Use
    </h3>
  </div>
  <section
    className="bcgov-scroll-box bcgov-printable"
  >
    <p>
      In these Terms of Use, "you" or "your" includes the individual using or accessing the Electronic Criminal Record Check Portal (the "Site") on their own behalf to submit an online request for a criminal record check.
    </p>
    <p>
      These Terms of Use are an agreement between you and Her Majesty the Queen in Right of the Province of British Columbia, represented by the Minister of Public Safety and Solicitor General (the "Province") and they govern your use of the Site and, where applicable, any associated service ("Associated Service" and, together with the Site, the "Services"). By clicking the box to indicate that you accept these Terms of Use, and in consideration of your use of the Services, you agree, to the terms and conditions set out below.
    </p>
    <p>
      Your failure to abide by these Terms of Use may result in the suspension or cancellation of your use of or access to the Services. In addition, the Province reserves the right to pursue any remedy available at law or in equity.
    </p>
    <p>
      Please print a copy of these Terms of Use for your records.
    </p>
    <h2>
      Disclaimer:
    </h2>
    <ol>
      <li>
        <b>
          Reasonable efforts have been made to provide accurate, complete and timely information regarding the Services and the Site in general. However, you are encouraged to refer to the 
          <i>
            Criminal Records Review Act
          </i>
          , any related Regulations, policy and other official information materials before submitting an online request for a CRC.
        </b>
      </li>
      <br />
      <li>
        The services are provided “as is”, without warranty of any kind including warranty of fitness for a particular purpose. Use of the services is entirely at your own risk and you will be liable for any failure to abide by these terms of use.
      </li>
      <li>
        Without limiting the general nature of the foregoing, the province does not represent or warrant that:
        <ol
          type="a"
        >
          <li>
            The accuracy, completeness or currency of services or any associated information, or that any errors will be corrected;
          </li>
          <li>
            That the services will function in a timely manner or will be available without error, failure or interruption; or
          </li>
          <li>
            That the services will meet your expectations or requirements.
          </li>
        </ol>
      </li>
      <li>
        The Province is not responsible for the content of the Payment Site as defined below.
      </li>
    </ol>
    <h2>
      Information Collection:
    </h2>
    <ol
      start="5"
    >
      <li>
        When you visit the Site or use the Services, certain types of information are automatically collected from you, through the use of audit logs or cookies. This information is collected, used and disclosed in accordance with the Province’s 
        <a
          href="https://www2.gov.bc.ca/gov/content/home/privacy"
          rel="noopener noreferrer"
          target="_blank"
        >
          Privacy Policy
        </a>
        .
      </li>
      <li>
        The date and time of your acceptance of these Terms of Use will be logged. This will enable you to skip this step on future visits. However, if these Terms of Use are modified, they will be presented to you upon your next following visit and you will need to accept the modified terms in order to continue to access the Services. Notwithstanding the foregoing, you are responsible for reviewing these Terms of Use on a regular basis to ensure that you are aware of any modifications that may have been made and your continued use of the Services constitutes your acceptance of any such modified Terms of Use.
      </li>
      <li>
        The information that you input on the Site may also be logged and attributed to you for verification purposes
      </li>
      <li>
        Any personal information that may be collected from you on this Site is collected, used and disclosed in accordance with the collection notice presented to you at the time of collection.
      </li>
    </ol>
    <h2>
      Authentication:
    </h2>
    <ol
      start="9"
    >
      <li>
        You are required to have a 
        <a
          href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card/login-with-card"
          rel="noopener noreferrer"
          target="_blank"
        >
          BC Services Card
        </a>
          to login and submit an online request for a criminal record check.
      </li>
      <li>
        Each time you access the Site, you must be authenticated by following an external link to the 
        <a
          href="https://www2.gov.bc.ca/gov/content/governments/government-id/bc-services-card"
          rel="noopener noreferrer"
          target="_blank"
        >
          BC Services Card Site
        </a>
        , following which you will be returned to, and allowed to access the application portions of, the Site.
      </li>
    </ol>
    <h2>
      Payment:
    </h2>
    <ol
      start="11"
    >
      <li>
        You will be required to access an external payment processing site ("Payment Site") in order to pay the criminal record check fee. For volunteers completing an online request for a criminal record check, no payment is required.
      </li>
    </ol>
    <h2>
      Additional Terms and Personal Information Collection:
    </h2>
    <ol
      start="12"
    >
      <li>
        You may be required to accept additional terms and conditions in order to use or access an Associated Service, including the BC Services Card Site, and/or the Payment Site, in which case your access to, and use of the services offered by, those Associated Services or the Payment Site is governed by such additional terms.
      </li>
      <li>
        Additional personal information may be collected from you by the providers of the Additional Sites in accordance with their terms and any associated privacy statement(s).
      </li>
    </ol>
    <h2>
      Acceptable Use and Security:
    </h2>
    <ol
      start="14"
    >
      <li>
        You must not:
        <ol
          type="a"
        >
          <li>
            use the Services for any unlawful or inappropriate purpose, including hacking, data mining or other intrusion activities;
          </li>
          <li>
            input or upload any information which contains viruses, Trojan horses, worms, time bombs or other computer programming routines that may damage or interfere with the performance or function of the Services or any Associated Service;
          </li>
          <li>
            divulge, share, compromise or permit any other person to use your login and password to access the Services;
          </li>
          <li>
            take any action that might reasonably be construed as altering, destroying, defeating, compromising or rendering ineffective the security related to the Site or any Associated Service, or being likely to affect other users of the Services;
          </li>
          <li>
            attempt to collect any information about other users of the Services; or
          </li>
          <li>
            decompile, disassemble, reverse engineer or otherwise copy any source code associated with the Site or any Associated Service.
          </li>
        </ol>
      </li>
    </ol>
    <h2>
      Ownership and Non-permitted Uses:
    </h2>
    <ol
      start="15"
    >
      <li>
        You acknowledge and agree that at all times the Province and/or the providers of the Additional Sites, or their respective licensors, are the owners of any software, hardware, servers, networks or other equipment used to provide the Services.
      </li>
      <li>
        You will not take any action that would be inconsistent with or infringe any proprietary or intellectual property rights of the Province, the providers of the Additional Sites or their respective licensors, in any software, hardware, servers, networks or other equipment, documentation or other information used to provide the Services.
      </li>
      <li>
        You will not remove or alter any proprietary symbol or notice, including any copyright notice, trademark or logo displayed in connection with the Services.
      </li>
    </ol>
    <h2>
      Suspension or Cancellation of Services:
    </h2>
    <ol
      start="18"
    >
      <li>
        Your use of any of the Services may be suspended or cancelled at any time if:
        <ol
          type="a"
        >
          <li>
            you fail to abide by these Terms of Use, or other terms and conditions that may be posted on any website used to access the Services; or
          </li>
          <li>
            the Province or the provider of any Associated Service deems such suspension or cancellation necessary for any good and valid reason.
          </li>
        </ol>
      </li>
      <li>
        The Province and any provider of any Associated Service reserve the right, at any time, to:
        <ol
          type="a"
        >
          <li>
            make changes to the Services;
          </li>
          <li>
            stop providing the Services; and
          </li>
          <li>
            modify these Terms of Use at any time, without notice being provided directly to you.
          </li>
        </ol>
      </li>
    </ol>
    <h2>
      Limitation of Liability:
    </h2>
    <ol
      start="20"
    >
      <li>
        In addition to the Province’s general 
        <a
          href="https://www2.gov.bc.ca/gov/content/home/disclaimer"
          rel="noopener noreferrer"
          target="_blank"
        >
          Limitation of Liabilities
        </a>
        , you agree that under no circumstances will the Province or the provider of any Associated Service be liable to you, a Representative or to any other individual or entity for any direct, indirect, special, incidental, consequential or other loss, claim, injury or damage, whether foreseeable or unforeseeable (including without limitation claims for damages for loss of profits or business opportunities, use of or inability to use the Services, interruptions, deletion or corruption of files, loss of programs or information, errors, defects or delays) arising out of or in any way connected with your or their access to or use of the Services or any failure by you or them to abide by these Terms of Use and whether based on contract, tort, strict liability or any other legal theory. The previous sentence will apply even if the Province or the provider of any Associated Service has been specifically advised of the possibility of any such loss, claim, injury or damage.
      </li>
    </ol>
    <h2>
      Enforceability and Jurisdiction:
    </h2>
    <ol
      start="21"
    >
      <li>
        If any term or provision of these Terms of Use is invalid, illegal or unenforceable, all other terms and provisions of these Terms of Use will nonetheless remain in full force and effect.
      </li>
      <li>
        All access to the Site or use of any Services or Associated Service will be governed by, and construed and interpreted in accordance with, the laws applicable in the Province of British Columbia, Canada.
      </li>
      <li>
        You hereby consent to the exclusive jurisdiction and venue of the courts of the Province of British Columbia, sitting in Victoria, for the hearing of any matter relating to or arising from these Terms of Use and/or your access to the Site or use of the Services or any Associated Service.
      </li>
    </ol>
  </section>
  <section
    className="bcgov-non-printable pt-2"
  >
    <label
      htmlFor="acceptTerms"
    >
      <input
        id="acceptTerms"
        onClick={[MockFunction]}
        type="checkbox"
      />
       
      <b>
        I accept these terms and conditions
      </b>
      <span
        className="bcgov-mandatory"
        id="asterisk"
      >
        *
      </span>
    </label>
  </section>
</div>
`;
